### ISSUE TITLE
Did you get similar accuracy in the paper?

### ISSUE THREAD

**Issue by HLu97** [2598Nitz/grasp_multiObject_multiGrasp#1]
Did you get similar accuracy in the paper?
How does "Negative Transformation of Cornell Images" appear in your code?
I see that the processing in your code is to limit the minimum depth to 0 and normalize.
How accurate can this process be?
Hope you can answer, thank you!

**Comment by HLu97** [comment]
I found another representation in the ggcnn code.
`img[r, c] = np.sqrt(x ** 2 + y ** 2 + z ** 2)`
This is different from using the depth value "z" directly, do you have any opinion?

**Comment by 2598Nitz** [comment]
>1. Did you get similar accuracy in the paper?
- The accuracy on test dataset(20% of total images) of Cornell images was pretty good. I didn't calculate the accuracy but I claim that it was higher than 90% on Cornell dataset. 

> 2. How does "Negative Transformation of Cornell Images" appear in your code?
- So basically the grayscale range for realsense camera was inverse of the cornell grayscale image(depth image). So, I inverted the Cornell depth image to make it consistent with realsense camera images. Here's a [link](https://stackoverflow.com/questions/19580102/inverting-image-in-python-with-opencv) that might help in how to use it in code.

>3.  I see that the processing in your code is to limit the minimum depth to 0 and normalize.How accurate can this process be?
- Since the neural net defined requires the three channels RGD to be 8 bit sized, the Depth image had to be compressed from original size to 8 bit sized value. 
- It might affect the accuracy as we are missing out on initial relative difference in depth values by compressing them.
- Solving this problem would require training a new model which takes 16 bit resolution RGD channels and as far as I know defining 16 bit resolution channels is uncommon in Covoulutional Neural Nets, so you won't be able to take advantage of transfer learning.

> 4. img[r, c] = np.sqrt(x ** 2 + y ** 2 + z ** 2). This is different from using the depth value "z" directly, do you have any opinion?
- I am not sure about this but whether one should use euclidean distance considering x, y and z or just use z value would depend on the depth image output format of one's camera. If the depth image output considers x and y distance between camera and points in image then better use x and y values for obtaining depth image of cornell dataset.
